### PR TITLE
Add AVX10 backend and improve AVX2 backend for AES-GCM

### PR DESCRIPTION
**Pull request by ripatel-fd**
Before merging, ensure all these are complete 

- [x] Import assembly blobs
- [x] Remove redundant OpenSSL code
- [x] Add backend switching
- [x] Add tests
- [x] Add licensing information
- [x] Squash commits
- [x] Review by Emanuele

Results in a QUIC decryption throughput improvement between 2x and 2.5x for Zen 5.
A Ryzen 9700X can now decrypt 123 Gbps MTU-size QUIC traffic per core.

### CODE REVIEW

File: src/ballet/aes/fd_aes_gcm.h [hunk 1]
```diff
@@ -78,33 +134,36 @@ fd_aes_256_gcm_init( fd_aes_gcm_t * aes_gcm,
...
+fd_aes_gcm_decrypt( fd_aes_gcm_t * aes_gcm,
+                    uchar const *  c,
+                    uchar *        p,
+                    ulong          sz,
+                    uchar const *  aad,
+                    ulong          aad_sz,
+                    uchar const    tag[ static 16 ] );
+
+#define FD_AES_GCM_DECRYPT_FAIL (0)
+#define FD_AES_GCM_DECRYPT_OK   (1)
```
Comment (0x0ece) [inline src/ballet/aes/fd_aes_gcm.h:166]: nit: they may go up, before FD_PROTOTYPE_START

File: src/ballet/aes/fd_aes_gcm.h [hunk 2]
```diff
@@ -27,45 +27,101 @@
...
-  fd_aes_gcm_init( aes_gcm, key, 24UL, iv );
-}
-
-static inline void
-fd_aes_256_gcm_init( fd_aes_gcm_t * aes_gcm,
-                     uchar const    key[ static 32 ],
-                     uchar const    iv [ static 12 ] ) {
-  fd_aes_gcm_init( aes_gcm, key, 32UL, iv );
-}
+                     uchar const    iv [ static 12 ] );
```
Comment (0x0ece) [inline src/ballet/aes/fd_aes_gcm.h:124]: nit: remove static?

File: src/ballet/aes/fd_aes_gcm_x86.c [hunk 3]
```diff
@@ -0,0 +1,341 @@
...
+                         uint const                 le_ctr[4],
+                         uchar const                ghash_acc[16],
+                         ulong                      total_aadlen,
+                         ulong                      total_datalen,
+                         uchar const                tag[16],
+                         int                        taglen );
+
+void
+fd_aes_128_gcm_init_aesni( fd_aes_gcm_aesni_t * aes_gcm,
+                           uchar const          key[ static 16 ],
```
Comment (0x0ece) [inline src/ballet/aes/fd_aes_gcm_x86.c:115]: remove static everywhere in this file

**Review by 0x0ece — APPROVED** [review]

File: src/ballet/aes/fd_aes_gcm.h (hunk 1 above)
Comment (ripatel-fd) [reply src/ballet/aes/fd_aes_gcm.h:166]: I'd rather keep them close to the function they relate to

File: src/ballet/aes/fd_aes_gcm.h [hunk 4]
```diff
@@ -27,45 +27,101 @@
...
+  uchar pad2[  52 ];
+};
+typedef struct fd_aes_gcm_aesni_state fd_aes_gcm_aesni_t;
+
+/* AVX10 backend internals ********************************************/
+
+struct fd_aes_gcm_avx10_state {
+  fd_aes_gcm_aesni_key_t key;
+  uchar pad1[  28 ];
+  uchar gcm [ 320 ];
```
[the remark this answers is not in the document]
Comment (ripatel-fd) [reply src/ballet/aes/fd_aes_gcm.h:59]: @0x0ece These are copied over from hardcoded offsets in the assembly blobs. Added comments explaining that

**Review by 0x0ece — APPROVED** [review]

**Pull request merged**